aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,652 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/mwaa"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/mwaa/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/mwaa/waiter"
+)
+
+func resourceAwsMwaaEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMwaaEnvironmentCreate,
+		Read:   resourceAwsMwaaEnvironmentRead,
+		Update: resourceAwsMwaaEnvironmentUpdate,
+		Delete: resourceAwsMwaaEnvironmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"airflow_configuration_options": {
+				Type:      schema.TypeMap,
+				Optional:  true,
+				Sensitive: true,
+				Elem:      &schema.Schema{Type: schema.TypeString},
+			},
+			"airflow_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dag_s3_path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_class": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "mw1.small",
+			},
+			"execution_role_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Nit: Might be good to add ARN validation here. 👍 

```suggestion
			"execution_role_arn": {
				Type:         schema.TypeString,
				Required:     true,
				ValidateFunc: validateArn,
			},
```